src/iris/api.py

```diff
@@ -701,7 +701,10 @@ def is_valid_tracking_settings(t, k, tpl):
                 except jinja2.TemplateSyntaxError as e:
                     return False, 'Invalid jinja syntax in email html: %s' % e
     else:
-        if t not in cache.modes:
+        cache.api_cache_lock.acquire()
+        mode = cache.modes.get(t)
+        cache.api_cache_lock.release()
```

Comment:
The Lock object should support acting like a context manager. We can replace the manual acquire/release calls with:

```python
with cache.api_cache_lock:
    mode = cache.modes.get(t)
```

Reply:
cool, I hadn't realized the gevent semaphore is a context manager and can be used in with statements